### PR TITLE
Bump serving core 1.2.0 with deps

### PR DESCRIPTION
**Pull request by skonto**
- replaces #1487 
- updates deps due to [this](https://github.com/openshift-knative/serverless-operator/pull/1487#issuecomment-1092826430). Also we cannot avoid bringing in deps due to [this](https://github.com/openshift-knative/serverless-operator/pull/1487#issuecomment-1092894971).
- aligns a few deps to keep the 4.6 (1.24.1) support, some may be adjusted/removed further when knative operator will be updated again to 1.2.0.
- updates csv with the new Serving images
- split in three commits to help the review due to the vendor size

/cc @nak3 @matzew @maschmid @aliok

### CODE REVIEW

**Comment by skonto** [comment]
Failures due to:

```
        Reason:Pulling
        Message:Pulling image "quay.io/openshift-knative/kafka-consumer:v1.2"
```

**Comment by pierDipi** [comment]
Done in https://github.com/openshift/release/pull/27716

**Comment by skonto** [comment]
First failure due to:
```
    revision_timeout_test.go:170: Failed to create Service: admission webhook "webhook.metadata-webhook.example.com" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "maxDurationSeconds"
```
Can't do much though because although I have upgraded the Service def the webhook image is not being built and used along with the PR. Here is the image [registry.ci.openshift.org/openshift/openshift-serverless-nightly:metadata-webhook](https://github.com/openshift-knative/serverless-operator/blob/d0337700e4aef89b1857221cb0563f2dcd74390e/serving/metadata-webhook/config/webhook.yaml#L35).

**Comment by skonto** [comment]
@nak3 @matzew gentle ping.

File: go.mod [hunk 1]
```diff
@@ -19,23 +19,23 @@ require (
...
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
-	k8s.io/api v0.21.4
-	k8s.io/apimachinery v0.21.4
-	k8s.io/client-go v0.21.4
-	knative.dev/eventing v0.27.1
+	k8s.io/api v0.22.5
+	k8s.io/apimachinery v0.22.5
+	k8s.io/client-go v0.22.5
+	knative.dev/eventing v0.29.0
 	knative.dev/eventing-kafka v0.27.0
```
Comment (matzew) [inline go.mod:26]: any reason these got not bumped, while other eventing libs got bumped?

File: hack/update-deps.sh [hunk 2]
```diff
@@ -13,19 +13,19 @@ set -o pipefail
...
 EVENTING_KAFKA_VERSION="release-v1.0"
 EVENTING_KAFKA_BROKER_VERSION="release-v1.0"
-SERVING_VERSION="release-v1.0.1"
+SERVING_VERSION="release-v1.2.0"
 
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/hack@${KN_VERSION}"
+  "knative.dev/hack@v0.0.0-20220224013837-e1785985d364"
```
[the remark this answers is not in the document]
Comment (matzew) [reply hack/update-deps.sh:25]: using tags sounds good

**Comment by matzew** [comment]
LGTM

let's see we get all tests to green, before moving on

File: go.mod (hunk 1 above)
Comment (skonto) [reply go.mod:26]: This is weird didnt touch eventing during upgrading deps.

**Comment by skonto** [comment]
> let's see we get all tests to green, before moving on

We cant get all of them green as the metadata webhook is not being built with the PR so it will not have the latest Serving APIs and thus the missing field will cause the upstream tests to fail ^^^.  /cc @nak3 what can we do about this?

**Comment by nak3** [comment]
Yeah, we need to replace the image with the temporary image (built by yourself) or just skip the upstream-mesh test.

**Comment by skonto** [comment]
@matzew let's just skip the upstream test failure as once this is merged we will get the image built and test again in the subsequent PRs.

File: hack/update-deps.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (skonto) [reply hack/update-deps.sh:25]: Yeah I will do that in another PR as soon as all deps are in.

**Comment by skonto** [comment]
Using `quay.io/skontopo/openshift-serverless:metadata-webhook` as a temp image to show that tests pass. We need to revert this once this PR is merged.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**